File: backend/src/main/java/com/monitor/backend/data/FieldMonitorSchedule.java
package com.monitor.backend.data;

public class FieldMonitorSchedule {
    private String adminId, projectId, fieldMonitorScheduleId,
            date, organizationId, fieldMonitorId;
    private String projectName, organizationName;
    private int perDay, perWeek, perMonth;

    public String getFieldMonitorScheduleId() {
        return fieldMonitorScheduleId;
    }

    public void setFieldMonitorScheduleId(String fieldMonitorScheduleId) {
        this.fieldMonitorScheduleId = fieldMonitorScheduleId;
    }

    public String getFieldMonitorId() {
        return fieldMonitorId;
    }

    public void setFieldMonitorId(String fieldMonitorId) {
        this.fieldMonitorId = fieldMonitorId;
    }

    public String getOrganizationId() {
        return organizationId;
    }

    public void setOrganizationId(String organizationId) {
        this.organizationId = organizationId;
    }

    public String getProjectName() {
        return projectName;
    }

    public void setProjectName(String projectName) {
        this.projectName = projectName;
    }

    public String getOrganizationName() {
        return organizationName;
    }

    public void setOrganizationName(String organizationName) {
        this.organizationName = organizationName;
    }

    public FieldMonitorSchedule() {
    }


    public String getAdminId() {
        return adminId;
    }

    public void setAdminId(String adminId) {
        this.adminId = adminId;
    }

    public String getProjectId() {
        return projectId;
    }

    public void setProjectId(String projectId) {
        this.projectId = projectId;
    }

    public String getDate() {
        return date;
    }

    public void setDate(String date) {
        this.date = date;
    }

    public int getPerDay() {
        return perDay;
    }

    public void setPerDay(int perDay) {
        this.perDay = perDay;
    }

    public int getPerWeek() {
        return perWeek;
    }

    public void setPerWeek(int perWeek) {
        this.perWeek = perWeek;
    }

    public int getPerMonth() {
        return perMonth;
    }

    public void setPerMonth(int perMonth) {
        this.perMonth = perMonth;
    }
}

File: monitorlibrary/lib/data/field_monitor_schedule.dart
import 'package:meta/meta.dart';

class FieldMonitorSchedule {
  String? fieldMonitorScheduleId,
      fieldMonitorId,
      adminId,
      organizationId,
      projectId,
      projectName,
      date,
      organizationName;
  int? perDay, perWeek, perMonth;

  FieldMonitorSchedule(
      {required this.fieldMonitorId,
      required this.adminId,
      required this.projectId,
      required this.date,
      required this.fieldMonitorScheduleId,
      this.perDay,
      this.perWeek,
      this.perMonth,
      this.projectName,
      this.organizationName,
      required this.organizationId});

  FieldMonitorSchedule.fromJson(Map data) {
    this.fieldMonitorId = data['fieldMonitorId'];
    this.fieldMonitorScheduleId = data['fieldMonitorScheduleId'];
    this.adminId = data['adminId'];
    this.organizationId = data['organizationId'];
    this.projectId = data['projectId'];
    this.projectName = data['projectName'];
    this.organizationName = data['organizationName'];
    this.perDay = data['perDay'];
    this.perWeek = data['perWeek'];
    this.perMonth = data['perMonth'];
    this.date = data['date'];
  }
  Map<String, dynamic> toJson() {
    Map<String, dynamic> map = {
      'fieldMonitorId': fieldMonitorId,
      'fieldMonitorScheduleId': fieldMonitorScheduleId,
      'date': date,
      'adminId': adminId,
      'organizationId': organizationId,
      'projectId': projectId,
      'projectName': projectName,
      'organizationName': organizationName,
      'perDay': perDay,
      'perWeek': perWeek,
      'perMonth': perMonth,
    };
    return map;
  }
}
